### PR TITLE
SAK-47629 samigo: Add incorrect answer filter to student feedback

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-47629

### CODE REVIEW

**Comment by jesusmmp** [comment]
Does this work with all types of questions (including calculated questions)?

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java [hunk 1]
```diff
@@ -3736,6 +3752,7 @@ public double getAnswerScoreMCQ(ItemGradingData data, Map publishedAnswerHash)
 		  return 0d;
 	  }
 	  else if (answer.getIsCorrect()){ // instead of using answer score Item score needs to be used here 
+          data.setIsCorrect(Boolean.TRUE);
```
Comment (jesusmmp) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java:3755]: Spacing

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java [hunk 2]
```diff
@@ -497,15 +498,28 @@ public Map getStudentGradingData(String assessmentGradingId)
...
       Long gradingId = null;
-      if (assessmentGradingId != null) gradingId = Long.valueOf(assessmentGradingId);
-      return PersistenceService.getInstance().
+      if (assessmentGradingId != null) {
+        gradingId = Long.valueOf(assessmentGradingId);
+      }
+      Map<Long, List<ItemGradingData>> data = PersistenceService.getInstance().
         getAssessmentGradingFacadeQueries().getSubmitData(Long.valueOf(publishedId), agentId, scoringoption, gradingId);
+
+      /*
```
Comment (jesusmmp) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java:512]: Remove comment block

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java [hunk 3]
```diff
@@ -635,6 +640,8 @@ protected AssessmentGradingData setAssessmentGradingFromItemData(DeliveryBean de
       ItemGradingData igd = (ItemGradingData) ( (List) itemGradingHash.get(
         keys.next())).toArray()[0];
       AssessmentGradingData agd = gradingService.load(igd.getAssessmentGradingId().toString(), false);
+      //Set itemGradingSet = gradingService.getItemGradingSet(agd.getAssessmentGradingId().toString()).stream().filter(gd -> !gd.getIsCorrect()).collect(Collectors.toSet());
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java:643]: Remove comment block

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java [hunk 4]
```diff
@@ -652,6 +652,10 @@ else if ("".equals(assessmentSettings.getRetractDateString())) {
 		if (StringUtils.isNotBlank(assessmentSettings.getFeedbackComponentOption()))
 		    feedback.setFeedbackComponentOption(new Integer(assessmentSettings.getFeedbackComponentOption()));
 
+		if (StringUtils.isNotBlank(assessmentSettings.getCorrectAnswerOption())) {
+		    feedback.setCorrectAnswerOption(new Integer(assessmentSettings.getCorrectAnswerOption()));
+        }
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:657]: Align

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SaveAssessmentSettings.java [hunk 5]
```diff
@@ -231,6 +231,11 @@ public AssessmentFacade save(AssessmentSettingsBean assessmentSettings, boolean
      feedback.setFeedbackDelivery(new Integer(assessmentSettings.getFeedbackDelivery()));
     if (StringUtils.isNotBlank(assessmentSettings.getFeedbackComponentOption()))
         feedback.setFeedbackComponentOption(new Integer(assessmentSettings.getFeedbackComponentOption()));
+
+    if (StringUtils.isNotBlank(assessmentSettings.getCorrectAnswerOption())) {
+        feedback.setCorrectAnswerOption(new Integer(assessmentSettings.getCorrectAnswerOption()));
+	}
```
Comment (jesusmmp) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SaveAssessmentSettings.java:237]: Align

**Comment by adrianfish** [comment]
@jesusmmp I put in this PR a little prematurely and should have marked it as draft. This is some work we're doing for a client and what I'd like to do is get the commit on their development server for them to test thoroughly, while this sits here as a draft PR. I still appreciate the review though, I'm just saying this so you know the various question types will get tested out properly. I've tested FIB and the various MCQ types, but I thought it would be best if the client's faculty get a chance to test it and ask for changes.

**Comment by jesusmmp** [comment]
> @jesusmmp I put in this PR a little prematurely and should have marked it as draft. This is some work we're doing for a client and what I'd like to do is get the commit on their development server for them to test thoroughly, while this sits here as a draft PR. I still appreciate the review though, I'm just saying this so you know the various question types will get tested out properly. I've tested FIB and the various MCQ types, but I thought it would be best if the client's faculty get a chance to test it and ask for changes.



> just saying this so you know the various question types will get tested out properl

Great. If I have some time next week, I'll test too!

**Comment by adrianfish** [comment]
@jesusmmp Calculated questions work. I just pushed a fix to make them work.

**Review by ern — APPROVED** [review]
Looks good

**Pull request merged**